### PR TITLE
Support scheduling adjustments for daylight savings by optionally allowing timezones in schedule tags

### PR DESCRIPTION
**Pull request by DavidIHunt**
We now support an optional timezone specification on schedule tags. E.g..

```start: 0 9 * * * *; stop: 0 17 * * * *```

can become

```start: 0 9 * * * *; stop: 0 17 * * * * | europe/london```

where `europe/london` is any IANA timezone. If a timezone is not provided we assume UTC.
This allows the scheduler to adjust for daylight savings observed in different regions of the world.

This also includes a new method on the scheduling service to allow us to get the expected state given a schedule and a date/time. This made it easier to test the timezone implementation and will allow us to eventually replace the old cronService.

**Note** this does not include any UI changes. That will be done in a separate PR as this one is already big enough IMO.

### CODE REVIEW

File: server/modules/scheduling/index.js [hunk 1]
```diff
@@ -159,25 +195,27 @@ function getScheduleForInstance(instance) {
 
 function parseEnvironmentSchedule(environmentSchedule) {
   if (environmentSchedule.ManualScheduleUp === false && environmentSchedule.ScheduleAutomatically === false) {
```
Comment (jake-cross) [inline server/modules/scheduling/index.js]: When two things like this act as a single predicate, I really like a single method here to explain what it means when these two things are the case. What single check are we performing here? *minor thing*

File: server/modules/scheduling/index.js [hunk 2]
```diff
@@ -195,10 +237,10 @@ function getTagValue(instance, tagName) {
 }
 
 function currentStateOfInstance(instance) {
-  if (instance.State.Name === 'running') return states.on;
-  if (instance.State.Name === 'stopped') return states.off;
+  if (instance.State.Name === 'running') return currentStates.on;
```
Comment (jake-cross) [inline server/modules/scheduling/index.js]: I like the renames!

File: server/test/modules/scheduling-expectedStates.spec.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    it('should switch off at 18:00:00+0100 when scheduled off at 18:00:00 London time in June', () => {
+      let schedule = 'stop: 0 18 1 6 * 2017 | Europe/London';
+      let dateTime = '2017-06-01T18:00:00+0100';
+
+      let determinedState = scheduling.expectedStateFromSchedule(schedule, dateTime);
+      expect(determinedState).to.equal(scheduling.states.off);
+    });
+  });
+});
```
Comment (jake-cross) [inline server/test/modules/scheduling-expectedStates.spec.js]: Very clean!

File: server/modules/scheduling/index.js (hunk 1 above)
Comment (DavidIHunt) [inline server/modules/scheduling/index.js]: Yes I agree. I didn't touch that code here but I'll do it anyway. Worth making that a bit easier to read.

**Comment by DavidIHunt** [comment]
Please do not merge yet.. I think I may have discovered an issue.

File: server/modules/scheduling/index.js (hunk 1 above)
Comment (DavidIHunt) [inline server/modules/scheduling/index.js]: Done.

**Pull request merged**